application/src/main/java/org/togetherjava/tjbot/commands/mathcommands/wolframalpha/WolframAlphaCommand.java

```diff
@@ -0,0 +1,374 @@
+            @NotNull SlashCommandEvent event) {
+        QueryResult result;
+        try {
+            Files.writeString(Path
+                .of("C:\\Users\\Abc\\IdeaProjects\\TJ-Bot-baseRepo\\application\\src\\main\\java\\org\\togetherjava\\tjbot\\commands\\mathcommands\\wolframalpha\\responsebody.xml"),
+                    response.body());
+            result = XML.readValue(response.body(), QueryResult.class);
+
+        } catch (IOException e) {
+            event.getHook()
+                .setEphemeral(true)
+                .editOriginal("Unexpected response from WolframAlpha API")
+                .queue();
+            logger.error("Unable to deserialize the class ", e);
+            return Optional.empty();
+        }
+
+        if (!result.isSuccess()) {
+            event.getHook()
+                .setEphemeral(true)
+                .editOriginal("Could not successfully receive the result")
+                .queue();
+
+            // TODO The exact error details have a different POJO structure,
+            // POJOs have to be added to get those details. See the Wolfram doc.
+            return Optional.empty();
+        }
+        return Optional.of(result);
+    }
+
+    @Deprecated
+    private @NotNull Optional<List<MessageAction>> createImages(SlashCommandEvent event,
+            QueryResult result) {
+        MessageChannel channel = event.getChannel();
+        List<MessageAction> messages = new ArrayList<>();
+        // For each probably much more readable.
+        try {
+            result.getPods()
+                .forEach(pod -> pod.getSubPods().stream().map(SubPod::getImage).forEach(image -> {
+                    try {
+                        String name = image.getTitle();
+                        String source = image.getSource();
+                        String extension = ".png";
+                        if (name.isEmpty())
+                            name = pod.getTitle();
+                        name += extension;
+                        BufferedImage img = ImageIO.read(new URL(source));
+                        ByteArrayOutputStream stream = new ByteArrayOutputStream();
+                        ImageIO.write(img, "png", stream);
+                        messages.add(channel.sendFile(stream.toByteArray(), name));
+                    } catch (IOException e) {
+                        event.getHook()
+                            .setEphemeral(true)
+                            .editOriginal(
+                                    "Unable to generate message based on the WolframAlpha response")
+
+                            .queue();
+                        logger.error("Failed to read image {} from the WolframAlpha response",
+                                image, e);
+                        throw new Error();
+                    }
+                }));
+        } catch (Error e) {
+            return Optional.empty();
+        }
+        return Optional.of(messages);
```

Comment:
```suggestion
        return messages;
```
When you're wrapping a Collection in an optional something is going wrong.

If no items are found, an empty collection should be returned instead of null.

Reply:
I have removed that method, this is resolved